sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -190,7 +190,7 @@ private[hive] class HiveClientImpl(
     // For this reason we cannot load the jars added by ADDJarsCommand because of class loader
     // got changed. We reset it to clientLoader.ClassLoader here.
     state.getConf.setClassLoader(clientLoader.classLoader)
-    SessionState.start(state)
+    shim.setCurrentSessionState(state)
```

Comment:
@HyukjinKwon @sunchao I changed it to `shim.setCurrentSessionState(state)` to make it simpler.

Reply:
@wangyum I wonder if this is a safe change. And why do you think this is making it simpler?

Reply:
Sorry for late replay. It is two questions.

> if this is a safe change

Yes. The difference is `sessionState.start` creating too many directories and Spark SQL do not need it.

> why do you think this is making it simpler?

`shim.setCurrentSessionState(state)` is simpler than `if (version != hive.v12) SessionState.setCurrentSessionState(state) else   SessionState.start(state) `.